Shared/Samples/Generate offline map with custom parameters/GenerateOfflineMapWithCustomParametersView.CustomParameters.swift

```diff
@@ -0,0 +1,189 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension GenerateOfflineMapWithCustomParametersView {
+    struct CustomParameters: View {
+        /// The action to dismiss the sheet.
+        @Environment(\.dismiss) private var dismiss
+        
+        /// The view model for the download offline map area view.
+        @ObservedObject var model: Model
+        
+        /// The area of interest.
+        var extent: Envelope
+        
+        /// A Boolean value indicating whether the job is generating an offline map.
+        @Binding var isGeneratingOfflineMap: Bool
+
+        /// The error shown in the error alert.
+        @State private var error: Error?
+
+        /// The min scale level for the output. Note that lower values are zoomed further out,
+        /// i.e. 0 has the least detail, but one tile covers the entire Earth.
+        @State var minScaleLevel = 0.0
```

Comment:
It seems like most of the properties in in this view could be private, at the very least, the state variable as they should almost always be private.